### ISSUE TITLE
Using API Manager and API Portal in HA mode

### ISSUE THREAD

**Issue by dokuboyejo** [Axway/Cloud-Automation#39]
So, the template suggested using a single replica for API Manager
Can HA be enabled for API Manager when using external clustered Cassandra

**Comment by cwiechmann** [comment]
I think it should work to have multiple API-Manager replicas, but you have to make sure session-stickiness is properly configured, as the user-session is stored in a local API-Manager cache. Stickiness must be configured based on the session cookies: APIMANAGERSESSION & APIMANAGERSTATIC - How this works depends on your Kubernetes-Platform. 

Using an external or internal Cassandra makes no difference here.

**Comment by dokuboyejo** [comment]
@cwiechmann  where would both cookies be set ... at the load balancer level ?
Also, would using `sessiogAffinity` be sufficient?

**Comment by dokuboyejo** [comment]
@cwiechmann  Looks like increasing the replicas leads to repeated restart of the login screen for the api manager

**Comment by cwiechmann** [comment]
That's the result if session stickiness is not working. In that case, requests are distributed across the available PODs and each has its own session.

If you share more details about your environment and the configuration you have in place, I might be able to help.

**Comment by dokuboyejo** [comment]
The ingress section looks like below

```
ingress:
    enabled: true
    name: api-manager
    annotations:
      kubernetes.io/ingress.class: alb
      alb.ingress.kubernetes.io/scheme: internal # internet-facing
      alb.ingress.kubernetes.io/target-type: ip
      alb.ingress.kubernetes.io/group.name: axway-apim
      alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS": 443}]'
      alb.ingress.kubernetes.io/backend-protocol: "HTTPS"
      alb.ingress.kubernetes.io/auth-session-cookie: APIMANAGERSESSION
      alb.ingress.kubernetes.io/certificate-arn: arn:aws:acm:xxxx
    tls:
      - secretName: xxx
        hosts:
          - api-manager.xxxx
```

**Comment by cwiechmann** [comment]
I made some tests with my AWS environment. I was not able to get it working using an Application-Based Cookie. But it works as expected, when using the Load-Balancer-Generated cookie. 

For that, I have added the following annotation to configure the Target-Group with Stickiness, which is hit by the Load-Balancer:
```
alb.ingress.kubernetes.io/target-group-attributes: stickiness.enabled=true,stickiness.type=lb_cookie,stickiness.lb_cookie.duration_seconds=43200
```
This configures stickiness for the Target-Group, where PODs are registered and the Load-Balancer is talking to like so:
![image](https://user-images.githubusercontent.com/16292002/165776444-bf0d3d89-15c2-45d7-b5ca-2c0c0b574072.png)

And the Stickiness for the target group is configured like so:
![image](https://user-images.githubusercontent.com/16292002/165776663-71dec0ae-1d40-4280-ace6-316c94a54667.png)

I have aligned the stickiness timeout with my API-Manager session timeout. Ultimately, you only have to make sure, the Target-Group Stickiness timeout is longer than the configured API-Manager Session-Timeout. If a session is expired in API-Manager the user will be redirected to the login screen anyway, but it must be avoided, that the user hits the incorrect API-Manager still using a valid session cookie.
